pkg/controller/clusterdeployment/clusterdeployment_controller.go

```diff
@@ -830,6 +854,61 @@ func (r *ReconcileClusterDeployment) setupClusterInstallServiceAccount(namespace
 	return currentSA, nil
 }
 
+func (r *ReconcileClusterDeployment) ensureManagedDNSZone(cd *hivev1.ClusterDeployment, cdLog log.FieldLogger) (bool, error) {
+	// for now we only support AWS
+	if cd.Spec.AWS == nil || cd.Spec.PlatformSecrets.AWS == nil {
+		cdLog.Error("cluster deployment platform is not AWS, cannot manage DNS zone")
+		return false, fmt.Errorf("only AWS managed DNS is supported")
+	}
+	dnsZone := &hivev1.DNSZone{}
+	zoneName := types.NamespacedName{Namespace: cd.Namespace, Name: dnsZoneName(cd.Name)}
+	logger := cdLog.WithField("zone", zoneName.String())
+
+	err := r.Get(context.TODO(), zoneName, dnsZone)
+	if err != nil && !errors.IsNotFound(err) {
+		logger.WithError(err).Error("failed to fetch DNS zone")
+		return false, err
+	}
+	if err != nil {
```

Comment:
Consider changing to:

* First check if `err == nil` and do the success case and return.
* Next check `errors.IsNotFound(err)` and do the not found case
* Last do the general error case.